Makefile-rpm.mk

```diff
@@ -0,0 +1,29 @@
+NAME    := raft
+SRC_EXT := gz
+SOURCE   = v$(VERSION).tar.$(SRC_EXT)
+
+GIT_SHA1        := $(shell git rev-parse HEAD)
+GIT_SHORT       := $(shell git rev-parse --short HEAD)
+GIT_NUM_COMMITS := $(shell git rev-list HEAD --count)
+
+GIT_INFO := $(GIT_NUM_COMMITS).g$(GIT_SHORT)
+
+BUILD_DEFINES := --define "%relval $(GIT_INFO)"
+RPM_BUILD_OPTIONS := $(BUILD_DEFINES)
+
+%.$(SRC_EXT): %
+	rm -f $@
+	gzip $<
+
+$(NAME)-$(VERSION).tar:
+	git archive --format tar --prefix $(NAME)-$(VERSION)/ -o $@ HEAD
+
+v$(VERSION).tar: $(NAME)-$(VERSION).tar
+	mv $< $@
+
+include packaging/Makefile_packaging.mk
+
+show_git_metadata:
+	@echo $(GIT_SHA1):$(GIT_SHORT):$(GIT_NUM_COMMITS)
```

Comment:
This should probably go into `packaging/Makefile_packaging.mk` also.

Reply:
Moved in next update.